src/SamplesApp/UITests.Shared/Microsoft_UI_Xaml_Controls/RatingControlTests/RatingControlPage.xaml.cs

```diff
@@ -0,0 +1,639 @@
+
+			CustomImagesTwoLoadedStageOneCheckBox.IsChecked = true;
+		}
+
+		private void TestRatingControl_Loaded(object sender, RoutedEventArgs e)
+		{
+			_dt = new DispatcherTimer();
+			_dt.Interval = TimeSpan.FromSeconds(0.1);
+			_dt.Start();
+			_dt.Tick += DispatcherTimer_Tick;
+
+			DependencyObject obj = FindVisualChildByName(TestRatingControl, "RatingBackgroundStackPanel");
+			UIElement ui = obj as UIElement;
+			DependencyObject child = VisualTreeHelper.GetChild(obj, 1);
+			TextBlock tb = child as TextBlock;
+			this._secondTextBlockUI = tb;
+
+			SetupAnimatedValuesSpy();
+			SpyAnimatedValues();
+		}
+
+		private void DispatcherTimer_Tick(object sender, object e)
+		{
+			SpyAnimatedValues();
+		}
+
+		// Composition property spy stuff below:
+		private CompositionPropertySet AnimatedValuesSpy
+		{
+			get;
+			set;
+		}
+
+		private ExpressionAnimation ScaleAnimation
+		{
+			get;
+			set;
+		}
+
+		private bool IsRenderingHooked
+		{
+			get;
+			set;
+		}
+
+		private uint UIThreadTicksForValuesSpy
+		{
+			get;
+			set;
+		}
+
+		private void SetupAnimatedValuesSpy()
+		{
+			StopAnimatedValuesSpy();
+
+			this.AnimatedValuesSpy = null;
+			this.ScaleAnimation = null;
+
+			if (this._secondTextBlockUI != null)
+			{
+				const string visualScaleTargetedPropertyName = "visual.Scale";
+				UIElement ppChild = this._secondTextBlockUI;
+				Visual visualPPChild = ElementCompositionPreview.GetElementVisual(ppChild);
+
+				Compositor compositor = visualPPChild.Compositor;
+
+				this.AnimatedValuesSpy = compositor.CreatePropertySet();
+
+				this.AnimatedValuesSpy.InsertVector3("Scale", new Vector3(0.0f, 0.0f, 0.0f));
+
+				this.ScaleAnimation = compositor.CreateExpressionAnimation(visualScaleTargetedPropertyName);
+				this.ScaleAnimation.SetReferenceParameter("visual", visualPPChild);
+
+				CheckSpyingTicksRequirement();
+
+				TickForValuesSpy();
+			}
+			else
+			{
+				ResetSpyOutput();
+			}
+		}
+
+		private void StartAnimatedValuesSpy()
+		{
+			if (this.AnimatedValuesSpy != null)
+			{
+				this.AnimatedValuesSpy.StartAnimation("Scale", this.ScaleAnimation);
+			}
+		}
+
+		private void StopAnimatedValuesSpy()
+		{
+			if (this.AnimatedValuesSpy != null)
+			{
+				this.AnimatedValuesSpy.StopAnimation("Scale");
+			}
+		}
+
+		private void SpyAnimatedValues()
+		{
+			if (this.AnimatedValuesSpy != null)
+			{
+				StopAnimatedValuesSpy();
+
+				Vector3 scale;
+				CompositionGetValueStatus status = this.AnimatedValuesSpy.TryGetVector3("Scale", out scale);
+				if (CompositionGetValueStatus.Succeeded == status)
+				{
+					this.ScaleTextX.Text = scale.X.ToString();
+					this.ScaleTextY.Text = scale.Y.ToString();
+				}
+				else
+				{
+					this.ScaleTextX.Text = this.ScaleTextY.Text = "status=" + status.ToString();
+				}
+
+				StartAnimatedValuesSpy();
+
+				// System.Diagnostics.Debug.WriteLine("Spied values: " + this.ScaleTextX.Text);
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6162540)